devito/dle/rewriters.py

```diff
@@ -0,0 +1,535 @@
+import abc
```

Comment:
in this file, only two methods have changes:

* _optimize_halospots
* _loop_blocking

You can safely ignore the rest